cpg_utils/hail_batch.py

```diff
@@ -78,235 +74,116 @@ def remote_tmpdir(hail_bucket: Optional[str] = None) -> str:
-
-    def full_path(self, prefix: str, suffix: str) -> str:
-        """Build full path from prefix and suffix"""
-        return os.path.join(f'{self.scheme}://', prefix, suffix)
-
-
-class AzurePathScheme(PathScheme):
-    """
-    Azure Blob Storage path scheme, following the Hail Batch hail-az format.
-    """
-
-    def __init__(self, account: Optional[str] = 'cpg'):
-        config = get_config()
-        self.scheme = 'hail-az'
-        self.account = config['workflow'].get('azure_account', account)
-
-    def path_prefix(self, dataset: str, category: str) -> str:
-        """Build path prefix used in dataset_path"""
-        return f'{self.account}/{dataset}-{category}'
-
-    def full_path(self, prefix: str, suffix: str) -> str:
-        """Build full path from prefix and suffix"""
-        return os.path.join(f'{self.scheme}://', prefix, suffix)
-
-
-class LocalPathScheme(PathScheme):
-    """
-    Local posix path scheme. Requires workflow/local_dir to be set.
-    Creates directories automatically (mimicking the cloud FS behaviour).
-
-    Useful only for tests, don't use in production.
-    """
-
-    def __init__(self):
-        if not (local_dir := get_config()['workflow'].get('local_dir')):
-            local_dir = tempfile.mkdtemp(prefix='cpg-utils-')
-        self.local_dir = to_path(local_dir)
-        self.local_dir.mkdir(exist_ok=True, parents=True)
-        self.scheme = 'local'
-
-    def path_prefix(self, dataset: str, category: str) -> str:
-        """Build path prefix used in dataset_path"""
-        return f'{dataset}-{category}'
-
-    def full_path(self, prefix: str, suffix: str) -> str:
-        """Build full path from prefix and suffix"""
-        path = self.local_dir / prefix / suffix
-        path.parent.mkdir(parents=True, exist_ok=True)
-        return str(path)
-
-
-class Namespace(Enum):
-    """
-    Storage namespace.
-    https://github.com/populationgenomics/team-docs/tree/main/storage_policies#main-vs-test
-    """
-
-    MAIN = 'main'
-    TEST = 'test'
-
-    @staticmethod
-    def from_access_level(str_val: str) -> 'Namespace':
-        """
-        Parse value from an access level string.
-        >>> Namespace.from_access_level('test')
-        Namespace.TEST
-        >>> Namespace.from_access_level('standard')
-        Namespace.MAIN
-        >>> Namespace.from_access_level('main')
-        Namespace.MAIN
-        """
-        for val, str_vals in {
-            Namespace.MAIN: ['main', 'standard', 'full'],
-            Namespace.TEST: ['test'],
-        }.items():
-            if str_val in str_vals:
-                return val
-        raise ValueError(f'Cannot parse namespace or access level {str_val}')
-
-
 def dataset_path(
     suffix: str,
-    category: Optional[str] = None,
-    dataset: Optional[str] = None,
-    access_level: Optional[str] = None,
-    path_scheme: Optional[str] = None,
+    category: str | None = None,
+    dataset: str | None = None,
+    test: bool = False,
```

Comment:
This looks like a breaking API change and should probably come with a major version bump? Have we checked whether there are many invocations that set `access_level`, which would be affected?

@illusional Didn't you prefer to have the `test` case be handled by a different method, so it's super clear that setting `test=True` breaks the portability of a pipeline and should only be used by CPG-internal pipelines?

Reply:
Yes, would still prefer separate method if that’s okay.

Reply:
Added a separate method